cmd/transfer-pvc/transfer-pvc.go

```diff
@@ -186,151 +254,177 @@ func (t *TransferPVCOptions) getClientFromContext(ctx string) (client.Client, er
 	return client.New(restConfig, client.Options{Scheme: scheme.Scheme})
 }
 
-func (t *TransferPVCOptions) getRestConfigFromContext(ctx string) (*rest.Config, error) {
+func (t *TransferPVCCommand) getRestConfigFromContext(ctx string) (*rest.Config, error) {
 	c := ctx
 	t.configFlags.Context = &c
 
 	return t.configFlags.ToRESTConfig()
 }
 
-func (t *TransferPVCOptions) run() error {
-	srcCfg, err := t.getRestConfigFromContext(t.SourceContext)
+func (t *TransferPVCCommand) run() error {
+	zaplog, err := zap.NewProduction()
 	if err != nil {
-		log.Fatal(err, "unable to get source config")
+		log.Fatal(err, "failed to initiate logger")
 	}
 
-	destCfg, err := t.getRestConfigFromContext(t.DestinationContext)
+	logger := zapr.NewLogger(zaplog)
+
+	srcCfg, err := t.getRestConfigFromContext(t.Flags.SourceContext)
 	if err != nil {
-		log.Fatal(err, "unable to get destination config")
+		log.Fatal(err, "unable to get source rest config")
 	}
 
-	srcClient, err := t.getClientFromContext(t.SourceContext)
+	srcClient, err := t.getClientFromContext(t.Flags.SourceContext)
 	if err != nil {
 		log.Fatal(err, "unable to get source client")
 	}
-	destClient, err := t.getClientFromContext(t.DestinationContext)
+	destClient, err := t.getClientFromContext(t.Flags.DestinationContext)
 	if err != nil {
 		log.Fatal(err, "unable to get destination client")
 	}
 
 	// set up the PVC on destination to receive the data
-	pvc := &corev1.PersistentVolumeClaim{}
+	srcPVC := &corev1.PersistentVolumeClaim{}
 	err = srcClient.Get(
 		context.TODO(),
 		client.ObjectKey{
-			Namespace: t.pvcConfig.namespace.source,
-			Name:      t.pvcConfig.name.source,
+			Namespace: t.PVC.Namespace.source,
+			Name:      t.PVC.Name.source,
 		},
-		pvc,
+		srcPVC,
 	)
 	if err != nil {
 		log.Fatal(err, "unable to get source PVC")
 	}
 
-	destPVC := t.getDestinationPVC(pvc)
+	destPVC := t.buildDestinationPVC(srcPVC)
 	err = destClient.Create(context.TODO(), destPVC, &client.CreateOptions{})
 	if err != nil && !errors.IsAlreadyExists(err) {
 		log.Fatal(err, "unable to create destination PVC")
 	}
 
-	pvcList, err := transfer.NewPVCPairList(
-		transfer.NewPVCPair(pvc, destPVC),
-	)
+	labels := map[string]string{
+		"app.kubernetes.io/name":          "crane",
+		"app.kubernetes.io/component":     "transfer-pvc",
+		"app.konveyor.io/created-for-pvc": srcPVC.Name,
+	}
+
+	e, err := createEndpoint(t.Endpoint, destPVC, labels, logger, destClient)
 	if err != nil {
-		log.Fatal(err, "invalid pvc list")
+		log.Fatal(err, "failed creating endpoint")
 	}
 
-	var e endpoint.Endpoint
-	switch t.Endpoint {
-	case endpointRoute:
-		e = createAndWaitForRoute(destPVC, destClient)
-	case endpointNginx:
-		e = createAndWaitForIngress(destPVC, destClient)
-	default:
-		log.Fatalf("unsupported endpoint type %s\n", t.Endpoint)
+	waitForEndpoint(e, destClient)
+
+	if _, err := e.IsHealthy(context.TODO(), destClient); err != nil {
```

Comment:
nit: line 317 not needed anymore if above returns error

Reply:
@alaypatel07 its still needed actually. The wait function polls. so maybe 319 is not needed

Reply:
sorry I meant "L319 not needed anymore if L317 returns error", +1

Reply:
@alaypatel07 updated